src/poll.rs

```diff
@@ -222,7 +222,28 @@ impl Poller {
                 if self.notified.swap(false, Ordering::SeqCst) {
                     // `notify` will have sent a notification in case we were polling. We weren't,
                     // so remove it.
-                    return self.notify.pop_notification();
+
+                    // Pipes on Vita do not guarantee that after `write` call succeeds, the
+                    // data becomes immediately available for reading on the other side of the pipe.
+                    // To ensure that the notification is not lost, the read side of the pipe is temporarily
+                    // switched to blocking for a single `read` call.
+                    #[cfg(target_os = "vita")]
+                    rustix::fs::fcntl_setfl(
+                        &self.notify.read_pipe,
+                        rustix::fs::fcntl_getfl(&self.notify.read_pipe)?
+                            & !rustix::fs::OFlags::NONBLOCK,
+                    )?;
+
+                    let notification = self.notify.pop_notification();
+
+                    #[cfg(target_os = "vita")]
+                    rustix::fs::fcntl_setfl(
+                        &self.notify.read_pipe,
+                        rustix::fs::fcntl_getfl(&self.notify.read_pipe)?
+                            | rustix::fs::OFlags::NONBLOCK,
+                    )?;
+
+                    return notification;
```

Comment:
I think this could be inlined into `pop_notification`.

Reply:
Done